### PR TITLE
chore(react): fixing console errors on react primitives unit tests

### PR DESCRIPTION
**Pull request by esauerbo**
#### Description of how you validated changes
React unit tests pass with no errors printed to console 

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/src/hooks/__tests__/useAuth.test.ts [hunk 1]
```diff
@@ -78,7 +78,7 @@ describe('useAuth', () => {
 
       const { result, waitForNextUpdate } = renderHook(() => useAuth());
 
-      await waitForNextUpdate();
+      await act(async () => await waitForNextUpdate());
```
Comment (calebpollman) [inline packages/react/src/hooks/__tests__/useAuth.test.ts:81]: Think `waitForNextUpdate` returns `Promise<void>`, does the callback need to return?

**Review by calebpollman — COMMENTED** [review]
Just left a question but LGTM other than that. Thanks for the cleanup!

File: packages/react/src/hooks/__tests__/useAuth.test.ts (hunk 1 above)
Comment (esauerbo) [reply packages/react/src/hooks/__tests__/useAuth.test.ts]: In some cases yes but took out the async code where it wasn't necessary

File: packages/react/src/primitives/Accordion/__tests__/Accordion.test.tsx [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import { render, screen } from '@testing-library/react';
+import { act } from '@testing-library/react-hooks';
```
Comment (calebpollman) [inline packages/react/src/primitives/Accordion/__tests__/Accordion.test.tsx:2]: I don't think there are any hooks being tested in this file. Prefer to import act  from `'@testing-library/react'`

File: packages/react/src/hooks/__tests__/useAuth.test.ts (hunk 1 above)
Comment (calebpollman) [reply packages/react/src/hooks/__tests__/useAuth.test.ts]: Think my question may have been lost in translation, an arrow function with curly brackets creates an implicit return:
```
const returnThing = () => 'thing';

console.log(returnThing());
// 'thing';
```

In the case of:

```
await act(async () => await waitForNextUpdate());
```

`act` do anything with the return of its callback, so the implicit return is extraneous

**Review by calebpollman — COMMENTED** [review]
Still seeing imports of `act` from _@testing-library/react-hooks_ in files that are not testing React hooks, but ready to approve once those are cleaned up!

File: packages/react/src/hooks/__tests__/useAuth.test.ts [hunk 3]
```diff
@@ -54,7 +54,9 @@ describe('useAuth', () => {
 
     const { result, waitForNextUpdate } = renderHook(() => useAuth());
 
-    await waitForNextUpdate();
+    await act(async () => {
+      await waitForNextUpdate();
+    });
```
Comment (calebpollman) [inline packages/react/src/hooks/__tests__/useAuth.test.ts:59]: Non-blocking but we can probably remove the `act` and `await` here if we remove the `async` declaration on line 52 (same with line 69)

**Review by calebpollman — APPROVED** [review]
🚢

File: packages/react/src/hooks/__tests__/useAuth.test.ts (hunk 3 above)
Comment (esauerbo) [reply packages/react/src/hooks/__tests__/useAuth.test.ts:59]: I tried that but then the result comes back undefined and the test fails

**Review by reesscot — APPROVED** [review]

**Pull request merged**